components/admin/lessons/AdminLessonNav.tsx

```diff
@@ -0,0 +1,80 @@
+import { toUpper } from 'lodash'
+import { uniq } from 'lodash/fp'
+import React, { useState, useMemo } from 'react'
+import { Tab, Col, Nav } from 'react-bootstrap'
+import styles from '../../../scss/adminLessonNav.module.scss'
+
+type NavItem = { value: string }
+type Tab = () => JSX.Element
+
+type NavItems = [NavItem, ...NavItem[]]
+type TabsType = [Tab, ...Tab[]]
+
+type Props = {
+  navItems: NavItems
+  tabs: TabsType
+}
+
+const validateProps = ({ navItems, tabs }: Props) => {
+  if (navItems.length !== tabs.length)
+    throw new Error(
```

Comment:
why did you design it this way, where it could be error prone?

I would have done something like 

```js
[{
  tabName: ...,
  tabComponent: ...
  }, ... ]
  ```
  
  This way we don't have to worry about this error right?

Reply:
Yeah, this is a better design.